packages/e2e-tests/specs/editor/blocks/navigation.test.js

```diff
@@ -52,6 +52,9 @@ async function updateActiveNavigationLink( { url, label } ) {
 		await page.type( 'input[placeholder="Search or type url"]', url );
 		// Wait for the autocomplete suggestion item to appear.
 		await page.waitForXPath( `//span[@class="block-editor-link-control__search-item-title"]/mark[text()="${ url }"]` );
+		// Navigate to the first suggestion.
+		await page.keyboard.press( 'ArrowDown' );
```

Comment:
This demonstrates another issue which existed previously, where even if the user hadn't yet selected a suggestion, pressing Enter while suggestions are merely present would select one of them, rather than the use the user's input verbatim. This is related to the behavior described in https://github.com/WordPress/gutenberg/pull/19490#issuecomment-573894849 regarding how using `type="submit"` would cause the `onClick` handler to be called when Enter is pressed anywhere in the form.

Reply:
I agree. I think we need to decide on the UX here before we proceed. Is it:

1. Type anything into box.
  - type any URL
  - hit `ENTER`
  - see submission added as link

2. Selection required (aka "validation")
  - type something 
  - hit ENTER - if suggestion then select the first one
  - if no suggestion then the entered text is "invalid" and we should show a message

Personally, if we go for #2 then we need to know our validation logic is _absolutely_ sound else we're restricting what can be added as a link. We'd also need a means to display a validation error warning within the UI to the user if the entered text is not considered valid input.

Reply:
For me, I don't think we should use a suggestion unless it is "selected", assuming that we already have this mechanism for suggestion selection.

What I don't feel as strongly about is whether we choose to have a suggestion be selected by default. For example, we could have the first suggestion be selected by default, which might be a search result or otherwise the raw URL if no other search results exist (so, still optimizing for <kbd>Enter</kbd> inputting raw URL without any other interaction).

If we choose not to have a suggestion be selected by default, then I think we should be comfortable that if there is no selected suggestion, and I've entered something like "Contact Us" which yields a result, pressing <kbd>Enter</kbd> should either treat it like a raw URL (even if it's not _technically valid_) or do nothing until the user makes a selection or cancels the prompt. I can see this being a case for having a default selection, since none of this sounds like a very good user experience.

Reply:
I'm inclined to consider this non-blocking. The changes in this pull request aren't intending to influence this behavior, outside the fact it removes incidentally `<button type="submit">` handling. I would agree with you that this is something we can try to explore separately if there's a desire that <kbd>Enter</kbd> has a specific way that it considers the suggestions currently presented.